Src/BootCamp.Chapter/Program.cs

```diff
@@ -1,12 +1,86 @@
 ﻿using System;
+using System.Globalization; // NumberStyles.Float, CultureInfo.InvariantCulture
 
 namespace BootCamp.Chapter
 {
     class Program
     {
         static void Main(string[] args)
         {
-            BootCampHW.Lesson3.Demo();
+            string userName = "", userSurname = "";
+            int userAge = 0;
+            float userWeight = 0f, userHeight = 0f;
```

Comment:
Instead of declaring and initializing them here so you can use ref later on you can use the out parameter.

`CheckForUserInput(string message, out TYPE varName)` initializes the variable, so you can delete 10-12.
Instead of using the ref keyword when calling the functions you do use out then.
As you do not use values which are valid right now you should replace ref with out.